app.py

```diff
@@ -0,0 +1,35 @@
+import json
+from flask import Flask, jsonify, request
+
+def filenamify_url(url_str):
+    return_val = url_str
+    return return_val.replace('http://','').replace('https://','').replace('/','!').replace('#','_').replace('?','_')
+
+
+app=Flask(__name__)
+
+@app.route("/rest/",methods=['GET','POST'])
+def index():
+    if request.method=='GET': 
+        return jsonify({"msg":"Hello Flask!"})
+    if request.method=='POST': 
+        #content_type = request.headers.get('Content-Type')
+        #if (content_type == 'application/json'):  OR 'text/plain' 
+        #data = response.json() OR data = request.get_json()
+        data=json.loads(request.data)
+        print("Got json data from POST " + str(data))
+        #tested with  curl -XPOST -H "Content-Type: application/json" -d@codemeta.json http://10.160.0.24/rest/
+        #codeRepository or @id values
+        value=data['codeRepository'] + ".codemeta.json"
+        filename=filenamify_url(value)
+        filepath="/tmp/out/" + filename
+        print("cleaned filename: " + filename)
+        with open(filepath, 'w', encoding='latin-1') as my_file:
```

Comment:
I chosen latin-1 because it offers more language characters such as vocals with accents (è,.. ) or diaeresis (ë,...)  and so on, that can be potentially used in fields like "description"
UTF-8, in this context, can only give some low level characters are very likely to don't be used

Reply:
Actually it's the other way round :) utf-8 is an encoding for unicode, which allows encoding everything that is in latin-1, latin-2, etc etc.. and much more), latin1 is an old encoding with limited codepoints (only 256). There's no reason to use it except for legacy purposes.

I share your argument of course; we indeed want people to use diacritics (but also cyrillic, greek, chinese, etc etc) in their descriptions. UTF-8 allows that.

Reply:
(I fixed this after merge in https://github.com/CLARIAH/tool-discovery/commit/a366d2576ee92ab386f363dd0d47b5f315ee1a23)